plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -184,21 +184,30 @@ fu_plugin_thunderbolt_add (FuPlugin *plugin, GUdevDevice *device)
 	}
 
 	vid = fu_plugin_thunderbolt_udev_get_id (device, "vendor", &error);
-	if (vid == 0x0) {
+	if (vid == 0x0)
 		g_warning ("failed to get Vendor ID: %s", error->message);
-		return;
-	}
+
 	did = fu_plugin_thunderbolt_udev_get_id (device, "device", &error);
-	if (did == 0x0) {
+	if (did == 0x0)
 		g_warning ("failed to get Device ID: %s", error->message);
-		return;
-	}
 
 	is_host = fu_plugin_thunderbolt_is_host (device);
-	vendor_id = g_strdup_printf ("TBT:0x%04X", (guint) vid);
-	device_id = g_strdup_printf ("TBT-%04x%04x", (guint) vid, (guint) did);
+
+	version = g_udev_device_get_sysfs_attr (device, "nvm_version");
+	if (is_host && ((version == NULL) || (vid == 0x0) || (did == 0x0))) {
```

Comment:
I just looked at the [kernel source](https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/drivers/thunderbolt/switch.c#n889), and the only other error we could get is `-EAGAIN`, which is [mapped](https://github.com/GNOME/glib/blob/master/gio/gioerror.c#L215) to `G_IO_ERROR_WOULD_BLOCK`. So, `NULL` and not `G_IO_ERROR_WOULD_BLOCK` means safe mode? Good enough?